xdem/dem.py

```diff
@@ -0,0 +1,163 @@
+        self.vref = vref
+        self.vref_grid = vref_grid
+        self.ccrs = ccrs
+
+        # trying to get vref from product name
+        if read_vref_from_prod:
+            self.__parse_vref_from_fn(silent=silent)
+
+
+    def __parse_vref_from_fn(self,silent=False):
+
+        """
+        Attempts to pull vertical reference from product name identified by SatImg
+        """
+
+        if self.product is not None:
+            vref = parse_vref_from_product(self.product)
+            if vref is not None and self.vref is None:
+                if not silent:
+                    print('From product name "'+ str(self.product)+'": setting vertical reference as ' + str(self.vref))
+                self.vref = vref
+            elif vref is not None and self.vref is not None:
+                if not silent:
+                    print('Leaving user input of ' + str(self.vref) + ' for vertical reference despite reading ' + str(
+                        vref) + ' from product name')
+            else:
+                if not silent:
+                    print('Could not find a vertical reference based on product name: "'+str(self.product)+'"')
+
+
+    def set_vref(self,vref_name=None,vref_grid=None,compute_ccrs=False):
+        """
+        Set vertical reference with a name or with a grid
+
+        :param vref_name: Name of geoid
+        :param vref_grid: PROJ DATA geoid grid file name
+        :param compute_ccrs: Whether to compute the ccrs (possibly reading pyproj-data grid file)
+        :type vref_name: str
+        :type vref_grid: str
+        :type compute_ccrs: boolean
+
+        :return:
+        """
+
+        #for names, we only look for WGS84 ellipsoid or the EGM96/EGM08 geoids: those are used 99% of the time
+        if isinstance(vref_name, str):
+            if isinstance(vref_grid, str):
+                print('Both a vertical reference name and vertical grid are provided: defaulting to using name only.')
+            if vref_name == 'WGS84':
+                self.vref_grid = None
+                self.vref = 'WGS84'  # WGS84 ellipsoid
+            if vref_name == 'EGM08':
+                self.vref_grid = 'us_nga_egm08_25.tif'  # EGM2008 at 2.5 minute resolution
+                self.vref = 'EGM08'
+            elif vref_name == 'EGM96':
+                self.vref_grid = 'us_nga_egm96_15.tif'  # EGM1996 at 15 minute resolution
+                self.vref = 'EGM96'
+            else:
+                raise ValueError(
+                    'Vertical reference name must be either "WGS84", "EGM96" or "EGM08". Otherwise, provide only'
+                    'a geoid grid from PROJ DATA: https://github.com/OSGeo/PROJ-data')
+        elif not isinstance(vref_grid, str):
+            raise ValueError('Vertical reference grid name must be PROJ DATA file name, '
+                             'such as: "us_noaa_geoid06_ak.tif" for the Alaska GEOID2006')
+        else:
+            self.vref = 'Unknown vertical reference name'
```

Comment:
Maybe in that case, if both arguments are given, I would use grid only, and set self.vref to whatever is provided by the user?
Unless it doesn't make sense for how vref is used later...

Reply:
Actually, here self.vref should be set to None, otherwise self.to_vref for example will fail.

Reply:
Changed default behaviour: grid has priority; also added a check that the grid exists (based on pyproj.datadir)